src/authorization/TKAuthorizationEngine.m

```diff
@@ -0,0 +1,141 @@
+//
+//  TKAuthorizationEngine.m
+//  TokenSdk
+//
+//  Created by Sibin Lu on 12/12/17.
+//  Copyright © 2017 Token Inc. All rights reserved.
+//
+
+#import "TKAuthorizationEngine.h"
+#import "TKBrowser.h"
+#import "TKTokenBrowser.h"
+#import "TKError.h"
+#import "TKLocalizer.h"
+#import "TKLogManager.h"
+#import "TKJson.h"
+
+@implementation TKAuthorizationEngine {
+    TKBrowserCreationBlock browserCreationBlock;
+    ExternalAuthorizationDetails *details;
+    OnSuccessWithBankAuthorization onSuccess;
+    OnError onError;
+    TKBrowser* browser;
+    
+    BOOL completionUrlIsFound;
+}
+
+/**
+ * The shared pool can prevent TKAuthorizationEngine being recycled unexpectedly by ARC
+ */
++ (id)sharedPool {
```

Comment:
Not sure why this is needed. If close or dismiss or revoke is always used to dismiss it, then there is a reference held by the client.